### ISSUE TITLE
Tabbing should work properly for iframed CC fields

### ISSUE THREAD

**Issue by alaca** [impress-org/givewp#5380]
## User Story

As an admin, I want that when tabbing through the on-site CC fields (Hosted Fields) the tab order of the fields should be such that the user can fill them out left-to-right and top-to-bottom in a logical order using the tab key, not reaching for their mouse.

## Details

Currently with Stripe, PayPal Donations, Square and others the "Cardholder Name" field is not properly allowing tabbing through the credit card fields.

## Expected Behavior
The tab order of the fields should be such that the user can fill them out left-to-right and top-to-bottom in a logical order. 

## Steps to Reproduce

1. Enable Stripe Credit Card payment gateway
2. Go to the Donation form page 
3. Select Donate with Credit Card option
4. Press the Tab key to navigate through the fields

## Visuals
![deepin-screen-recorder_Select area_20201015144605](https://user-images.githubusercontent.com/4222590/96128128-3ee67880-0ef5-11eb-998a-f65162b68ce8.gif)

## Acceptance Criteria

-   [ ] User can fill CC fields in a logical order using the tab key for Stripe - Credit Card PG
-   [ ] User can fill CC fields in a logical order using the tab key for Paypal Donations PG
-   [ ] User can fill CC fields in a logical order using the tab key for Square PG

**The Acceptance Criteria is not complete, I will update accordingly.**

**Comment by JasonTheAdams** [comment]
@alaca I notice that you listed out the payment gateways in the details, but the acceptance criteria is still a single line. I suggest breaking apart the acceptance criteria so it's one checkbox per payment gateway. That way we're confident each one has been tested and is working properly. 😃

**Comment by kevinwhoffman** [comment]
I just reviewed the markup of the multi-step form, and I think part of the reason we are having trouble with `tabindex` is because we have explicitly set `tabindex=1` on several inputs (e.g. First Name, Last Name, Email).

Ideally we would not use `tabindex` at all and instead order the form elements in a logical sequence so that the browser can naturally move from one focusable element to the next.

As noted on [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex):
> Avoid using tabindex values greater than 0. Doing so makes it difficult for people who rely on assistive technology to navigate and operate page content. Instead, write the document with the elements in a logical sequence.

So rather than adding even more `tabindex` attributes, I'd recommend seeing if we can actually remove our current use of `tabindex` and let the browser handle the tabbing order based on the order of our markup.

**Comment by kevinwhoffman** [comment]
@alaca I think I've narrowed down the cause of the problem. The tab order issue only occurs when Stripe is NOT the default payment option. When Stripe is not the default, this means that rendering of the Stripe Elements (iframes) is delayed until the user actually selects the credit card payment option. At this point the Stripe Elements display, however the tab order skips over them.

See my [screencast](https://cln.sh/GresdT) for more info.

**Comment by alaca** [comment]
@kevinwhoffman the problem is not only related to the Stripe PG, but to all Payment Gateways which are not set as default PG. Why tabbing works when Stripe is set as default PG is because the input fields of the default gateway are loaded immediately, the rest of PGs are loaded via ajax when you click on the radio button to choose another PG. 
The function responsible for setting the tabindex is this [one](https://github.com/impress-org/givewp/blob/826fbbbf6dff371429940a62d09a102e1a8e2083/src/Views/Form/Templates/Sequoia/assets/js/form.js#L625). Basically it sets the tabindex attribute of each input element on the `give_gateway_loaded` event which is fired when PG is successfully loaded via ajax. At that moment the loaded HTML is not yet in the DOM, and that's why tabindex is not set for those fields. 

Try this, set Ofline Donations as default PG. Navigate to the Donation form and load the Stripe Credit Card PG. Try tabbing through the fields. Now go back one step, and after that go forward one step. Try the tab key again. Now it should work.

**Comment by kevinwhoffman** [comment]
@henryholtgeerts Can you provide some context here regarding why we are explicitly setting `tabindex` in the multi-step form?

https://github.com/impress-org/givewp/blob/826fbbbf6dff371429940a62d09a102e1a8e2083/src/Views/Form/Templates/Sequoia/assets/js/form.js#L620-L632

**Comment by henryholtgeerts** [comment]
@kevinwhoffman Believe this route was taken to account for the back button (top left) and navigation buttons (along bottom of the form), and to ensure that focus moves to the correct element from when navigating via the navigation buttons or back button.

**Comment by alaca** [comment]
This is what I get when I remove the `setupTabOrder` function. I used Tab key to navigate through the form elements.
![deepin-screen-recorder_Select area_20201016193155](https://user-images.githubusercontent.com/4222590/96290384-c6f87b00-0fe6-11eb-84ab-f1ad056d0206.gif)

We could try wrapping the `setupTabOrder` function with `setTimeOut` function, and set 300 milliseconds delay, enough for the loaded elements to be inserted into DOM. I think this will solve the tabbing issue we have now.
